homeassistant/components/goodwe/number.py

```diff
@@ -0,0 +1,102 @@
+"""GoodWe PV inverter numeric settings entities."""
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import logging
+
+from goodwe import Inverter, InverterError
+
+from homeassistant.components.number import NumberEntity, NumberEntityDescription
+from homeassistant.const import ENTITY_CATEGORY_CONFIG, PERCENTAGE, POWER_WATT
+from homeassistant.helpers.entity import DeviceInfo
+
+from .const import DOMAIN, KEY_DEVICE_INFO, KEY_INVERTER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass
+class GoodweNumberEntityDescription(NumberEntityDescription):
+    """Class describing Goodwe sensor entities."""
+
+    getter: Callable[[Inverter], Awaitable[int]] | None = None
```

Comment:
Both getter and setter attributes should be required.

Reply:
Done